### ISSUE TITLE
[Feature Proposal] Muscle memory-friendly multiple pies with instant access

### ISSUE THREAD

**Issue by mockballed** [markusfisch/PieLauncher#129]
Here's a straightforward way to have different pie menus that can be used without having to look and without needing to consider where to tap on the screen:

As per usual, tap and hold brings up the main pie menu.
Or...
Double-tap and hold to bring up the second pie menu.
Or...
Triple-tap and hold to bring up the third pie menu.
And so on.

This behavior would be compatible with the default "Tap on home screen" option to open the drawer. Only a small delay before displaying the drawer would be needed to account for subsequent taps.

On the visual side, appearing pie menus could push the previous menu outwards while its icons fade out. 

I'm trying to come up with an elegant way to assign apps to different pies. Maybe something along the lines of dragging icons towards the left or right screen edge to switch to another pie menu configuration screen?

**Comment by mockballed** [comment]
Personally, I think having 3 pie menus with 8 apps each would be great. I'd set them up so that for any given app in the main pie menu, the corresponding apps in the same position on the secondary and tertiary menus would match it thematically.

For example: if my music app is to the right (tap & swipe right) then I'd open my podcast app with double-tap & swipe right; my voice recorder app with triple-tap & swipe right.

I've been enjoying PieLauncher since the moment it joined F-Droid and I don't think I can go back to a conventional launcher!

**Comment by markusfisch** [comment]
Hm, that sounds interesting 🤔

Currently, double (or multiple) tapping would collide with opening the app drawer, but that could be changed with a preference, of course. For example, the app drawer could open on a long press only. Although that would make launching apps that are not in the pie even slower. On the other hand there's already an option to open the app drawer with a dedicated icon in the pie, which would already work with double tapping.

About delays: I already played with using delays in https://github.com/markusfisch/PieLauncher/issues/110 which wasn't feeling very good. Even slight delays are quite noticeable, and can quickly become annoying. But I think a delay may not be necessary.

Will try… 😉

**Comment by mockballed** [comment]
Great!

True that. Even a fraction of a second waiting for the drawer after a tap would take away from the snappiness of it.

Maybe they are not mutually exclusive. The drawer could start popping up the moment the first tap is released, and if a second tap is registered within a small time window, the popup animation could "morph" into showing the second pie menu instead.

From a UX perspective, it could be interpreted as a tap and release to display all app icons, followed by an immediate tap & hold to rearrange a subset of them into another pie menu.

Count with me for testing. I'll try to get PieLauncher to compile so I can try stuff.

**Comment by devilsclaw** [comment]
The other way to have multiple pie menus would be to have sub pie menus as icons and when you hold over one for lets say 1.5 seconds it would then replace the current one with a new circle in the same position. I could offset the center to where your thumb currently is but that might make it hard to access depending where the icon was when hovering over it.

**Comment by MarshCastle** [comment]
> "...have sub pie menus as icons and when you hold over one ... it would then replace the current one with a new circle in the same position."

This sounds brilliant. I currently use 12 slices in a pie, and desperately need more! 

The app draw is awkward for me, so a second layer would help loads. 

Of topic, but sorting the app drawer would be an easy fix for much of my woes... Maybe I should open another thread for that one though 🤔

**Comment by MarshCastle** [comment]
I couldn't disagree more.

To me this is the equivalent of saying that more than one level of menu should be abandoned in favour of a desktop full of shortcuts.

To me, muscle memory of two gestures is no different than muscle memory of one. I can't see how 'Right then Down' would affect muscle memory at all. After all most gaming console controls use combinations. Where's the evidence that two motions Will have a derogatory effect on muscle memory?

**Comment by MarshCastle** [comment]
I don't get it.

I can't see 'bad" design choice - just different a different one...

**Comment by mockballed** [comment]
Both approaches are not mutually exclusive. Implementing pie-menu icons wouldn't force users to use them.

To clarify, the original post was not suggesting a hover-and-open approach, where pausing over a pie-menu icon opens another pie menu. It proposed a tap-and-hold gesture, where tapping the screen (touch and release) quickly followed by a touch-and-hold would trigger the secondary pie menu.

**Comment by mockballed** [comment]
Yes! PieLauncher is about being lazy: launching apps without even having to glance at the screen.
I'll pass on the keyboard typing. I'll have my pies and eat them, too.

**Comment by MarshCastle** [comment]
I still don't see a multiple pie mess.
Two gestures, on after the other that prevent having to look at the screen, can be carried out with one hand easily, and can give users that want it multi-gesture app access would be an advantage for me, and (it seems) to others in the thread.

Also... I personally believe in data sovereignty, as I don't think that data farming is good for humanity, and can't support it - so Google products like GBoard are absolutely out of the question.

**Comment by MarshCastle** [comment]
This has become a bit of a silly convo. You're repeating your beliefs as fact, and using your inability to remember how you laid out your menu to judge everyone else's abilities. It doesn't make sense to me.

I have twelve directions, and it took a week or two. Please don't assume the abilities of others or base them on your own.

This thread is for many people to discuss the points raised, and for @markusfisch or other contributors to decide what they think. It's becoming hijacked by this, and that's not ok. It wastes people's time, and disrespects their efforts.

For those reading, please accept my Apologies to all for my getting dragged in and elongating the conversation in a non-prductive fashion.

**Comment by devilsclaw** [comment]
Muscle memory is not based off instinct. that is one the the dumbest things I have ever heard. Muscle is purely repetition of the same action until it is basically done automatically with out much thinking.

**Comment by hmmmIndeed** [comment]
> Imagine you've trained yourself a lot with the current pie, and then you wanted to have an another pie just for more shortcuts. And then you realise that your training is messed up completely because there are now other set of shortcuts in the same exact locations.

The sliding direction isn't the only thing that'll be saved by muscle memory. For example, sliding my finger to the right and tapping once before I slide my finger to the right are two completely different muscle memories. Yes, they share the movement of sliding to the right, but they are two separate sets of motions.

Personally, I think the extra taps are a pretty good way to bring out other pie menus.

(sorry if this convo is long over, I just wanted to add my support for this idea since I think it's a good one)

**Comment by snan** [comment]
> The other way to have multiple pie menus would be to have sub pie menus as icons and [...] offset the center to where your thumb currently is 

That is how pie menus hove traditionally worked on other systems 👍🏻

Also back then you could let go before the icons were even visible, just subtle gestures. Like left then right might be one thing, just up another. The ring was tighter and the appearance of the icons was delayed while the input reading was instant so it felt like controlling the computre magically through gestures. The selling point was "using it normally teaches you the shortcut", unlike bar menus where the keyboard shortcut has not much in common with selecting a menu entry. Here, my hands learn that swipe up means Termux or whatever.

**Comment by snan** [comment]
> Imagine you've trained yourself a lot with the current pie, and then you wanted to have an another pie just for more shortcuts. And then you realise that your training is messed up completely because there are now other set of shortcuts in the same exact locations.

Good points (and yay to eating plants) in the old pie app I used to use (it was a browser extension in the days of one-button mouse), having subpies was optional. So if you're happy with your current setup, it doesn't have to change.
